Library/Homebrew/install.rb

```diff
@@ -15,6 +15,7 @@ module Install
     module_function
 
     def perform_preinstall_checks(all_fatal: false, cc: nil)
+      MacOS.symlink_current_pkgconfig_directory
```

Comment:
Could maybe just make this a `attempt_pkgconfig_directory_symlink` below `attempt_directory_creation`?

Reply:
Would that not restrict it to being run only inside `install.rb`, instead of also during `brew update`?

Reply:
Yes. I actively think it should not be run by `brew update` 😁. Could we try not doing it in `brew update` and ship a follow-up PR to add it there (with `critical` and self-merge without review if necessary) if we actually get any user reports that the `brew update` addition will fix?

Reply:
I think the opposite: it's weird, and probably wrong, to not do this during `brew update`.

The symlink is part of the internal state of `brew`; it even lives inside `HOMEBREW_LIBRARY`. A `brew update` should not leave that state out of sync.

Reply:
> The symlink is part of the internal state of brew; it even lives inside HOMEBREW_LIBRARY. A brew update should not leave that state out of sync.

What actually breaks if this is out of sync outside of installations?